src/zope/sendmail/delivery.py

```diff
@@ -103,16 +103,27 @@ def newMessageId(self):
         return "%s@%s" % (left_part, gethostname())
 
     def send(self, fromaddr, toaddrs, message):
-        parser = email.parser.Parser()
-        msg = parser.parsestr(message)
-        messageid = msg.get('Message-Id')
+        # Switch the message to be bytes immediately, any encoding
+        # peculiarities should be handled before.
+        if message is None:
+            header = b''
+        else:
+            if not isinstance(message, bytes):
+                message = message.encode('utf-8')
+            header = message.split(b'\n\n')[0]
```

Comment:
Here is a potential likely minor problem: if the message strictly follows RFC 822, it will use `\r\n` rather than `\n` line separators. The effect would be that the whole message would be classified as header - not indented but likely not serious as well.